src/ocaml/typing/shape.mli

```diff
@@ -151,7 +151,7 @@ module Make_reduce(Context : sig
 
     val find_shape : env -> Ident.t -> t
   end) : sig
-  val reduce : Context.env -> t -> t
+  val reduce : ?weak:bool -> Context.env -> t -> t
```

Comment:
I would rather expose two functions, `reduce` (or `strong_reduce`) and `weak_reduce`, because they do fairly different things.

Reply:
Thanks for having a look!